File: PHP/7php 2/specialNumber.php
<?php
/*
|--------------------------------------------------------------------------
| Special Number (Special Numbers Series #5)
|--------------------------------------------------------------------------
|
*/

function specialNumber($n) {
   $x = [0, 1, 2, 3, 4, 5];
   $num = strval($n);
   $count = 0;
   for ($i = 0; $i < strlen($num); $i++) {
      for ($j = 0; $j < count($x); $j++) {
         if ($num[$i] == $x[$j]) {
            $count++;
         }
      }
   }
   return $count == strlen($num) ? "Special!!" : "NOT!!";
}


function specialNumber($n) {
   return preg_match('/^[0-5]+$/', $n) ? "Special!!" : "NOT!!";
}


function specialNumber($n) {
   return strlen(str_replace(range(0,5), '', $n)) == 0 ? "Special!!" : "NOT!!";
}


function specialNumber(int $n): string {
   return max(str_split($n)) <= 5 ? 'Special!!' : 'NOT!!';
}


print_r(specialNumber(2) . " Special!!
");
print_r(specialNumber(55) . " Special!!
");
print_r(specialNumber(26) . " NOT!!
");
print_r(specialNumber(9) . " NOT!!
");

File: Python/7py 3/special_number.py
"""
|--------------------------------------------------------------------------
| Special Number (Special Numbers Series #5)
|--------------------------------------------------------------------------
|
"""

def special_number(number):
   x = '012345'
   num = str(number)
   count = 0
   for i in range(len(num)):
      for j in range(len(x)):
         if num[i] == x[j]:
            count+=1
   return "Special!!" if count==len(num) else "NOT!!"


def special_number(n):
   return "Special!!" if max(str(n)) <= "5" else "NOT!!"


def special_number(n):
   return 'Special!!' if all(int(e) < 6 for e in str(n)) else 'NOT!!'


def special_number(number):
   return ('Special' if set(str(number)) <= set('012345') else 'NOT') + '!!'


print(special_number(5), "Special!!")
print(special_number(9), "NOT!!")
print(special_number(23), "Special!!")
print(special_number(79), "NOT!!")